### ISSUE TITLE
Why the resource is not public accessible?

### ISSUE THREAD

**Issue by pensiero** [zendframework/zend-cache#12]
Why the method [getRedisResource()](https://github.com/zendframework/zend-cache/blob/master/src/Storage/Adapter/Redis.php#L81) is protected and not public?

In my case, I need to obtain the Redis resource and pass it to some modules (in this case, [DoctrineORM](http://doctrine-orm.readthedocs.org/en/latest/reference/caching.html#redis)).

**Comment by weierophinney** [comment]
It's protected likely as it was only intended for internal purposes.

That said, you have 3 options:

- Extend the class and make the method public:
```php
public function getRedisResource()
{
    return parent::getRedisResource();
}
```
  If you go this route, you'll also need to provide a custom factory to ensure your own adapter is used.
- Use reflection to make the method accessible, and then call it:
```php
$r = new ReflectionMethod($adapter, 'getRedisResource');
$r->setAccessible(true);
$resource = $r->invoke($adapter);
 ```
- Send a pull request that changes the visibility of the method.

**Comment by pensiero** [comment]
Thanks a lot @weierophinney, I'll send a PR and meanwhile I use the second method :+1:

**Comment by pensiero** [comment]
@weierophinney do you think it would be nice to change to public the same method in each adapter? (Redis/Memcached/Apc...) ?

**Comment by weierophinney** [comment]
Yes; it's useful to be able to share the adapters, and we cannot assume
everything is configured under our own service manager.

On Tue, Jun 16, 2015 at 9:52 AM, Oscar Fanelli <notifications@github.com>
wrote:

> @weierophinney <https://github.com/weierophinney> do you think it would
> be nice to change to public the same method in each adapter?
> (Redis/Memcached/Apc...) ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-cache/issues/12#issuecomment-112459807>
> .
>



-- 
Matthew Weier O'Phinney
matthew@weierophinney.net
https://mwop.net/

**Comment by marc-mabe** [comment]
@pensiero @weierophinney The resources of all resource based storages **are** already public accessible from the resource manager. `$storage->getOptions()->getResourceManager()->getResource()`

The resource manager is where you configure and share resources.

**Comment by pensiero** [comment]
@marc-mabe yes, but you can't access to the resource manager from the adapter

**Comment by marc-mabe** [comment]
You can as seen in my example above. The $storage variable holds the adapter instance.

**Comment by pensiero** [comment]
mmmmmm i'm not so skilled to say that this could be a better way or not... @weierophinney and others, what do you think about it?

**Comment by Maks3w** [comment]
@marc-mabe give a hint about to have a common API for `getResource()` as discussed in https://github.com/zendframework/zend-cache/pull/13

Could we just write a proxy method in the adapter?

```php
// Adapter.php
public function getResource() {
  return $this->getOptions()->getResourceManager()->getResource();
}
```

**Comment by marc-mabe** [comment]
@Maks3w Indeed it would be pretty simple to write a proxy method this is in the adapter and that's what @pensiero does in #13. I only figured out that he was though

**Comment by marc-mabe** [comment]
Closed as discussed in #13
